lms/djangoapps/teams/views.py

```diff
@@ -353,7 +355,10 @@ class TeamsListView(ExpandableFieldViewMixin, GenericAPIView):
     """
 
     # OAuth2Authentication must come first to return a 401 for unauthenticated users
-    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    # use OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser
+    # instead of OAuth2Authentication, SessionAuthentication to allow inactive user to get
+    # or create course team
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser)
```

Comment:
**Please consider this suggestion avoid this repetition of comments and authentication_classes**

Make a new file in `teams` named `mixins.py` 

```
from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
from openedx.core.lib.api.authentication import OAuth2AuthenticationAllowInactiveUser

class AllowInActiveUserAuthClassesMixin:
	# OAuth2AuthenticationAllowInactiveUser must come first to return a 401 for unauthenticated users
	# use OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser
    # instead of OAuth2Authentication, SessionAuthentication to allow inactive user to get
    # or create course team
    authentication_classes = (OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser)
```

2 new imports will be replaced with mixin import
Add AllowInActiveUserAuthClassesMixin in relevant classes
remove authentication_classes line from these classes which have AllowInActiveUserAuthClassesMixin


Or find some better way to avoid this repetition.

Reply:
@zeshanarifios I had a looked this is the pattern that has been followed through out edx code even in the case of more than 3 imports simultaneously, as I don't see any performance issues do you think we should deviate away from the standard pattern?

Reply:
Our pattern is if we are using same lines multiple times, we try to find some way to avoid this repetition and reuse the code. In my opinion we can should write and reuse `AllowInActiveUserAuthClassesMixin`.

@ahmed-faraz46 what are your thoughts on it ?

Reply:
In my case there is only one line change in core, in your suggested case there will be min of 2 lines change.One line would needed to be removed and the other would be adding a mixin.

Reply:
changes are done.